common.go

```diff
@@ -176,11 +177,52 @@ func sum512Truncate256(in []byte) [32]byte {
 	return sliceToByte32(sum512[:32])
 }
 
-func computePayloadHash(headerHash headerHash, nonce Nonce, payloadCiphertext []byte) payloadHash {
+func checkCiphertextState(version Version, ciphertext []byte, isFinal bool) error {
```

Comment:
This function is just for sanity checking our encryptor, is that right? Maybe add a comment about it?